### DISCUSSION TITLE
Composer plugins configuration

### DISCUSSION THREAD

**Discussion by s-shpachynskyi** [composer/composer#11944] [Q&A / Support] [answered]
Hello, everyone!
I have the following problem with installing packages on different git branches. 
Let's assume I have `feature1` and `feature2` branches. On `feature2` branch I install a new package [endroid/qr-code-bundle](https://github.com/endroid/qr-code-bundle), which uses composer plugin [endroid/installer](https://github.com/endroid/installer) and everything works fine here. Then when I switch to `feature1` branch and need to execute composer install, I can see a warning about the plugin which is not configured in composer.json, like on screenshot: 

![screenshot](https://github.com/composer/composer/assets/138432375/055d797a-e426-4269-80ac-9f771b4de70c)

But on `feature1` branch, there is no information in composer.json or composer.lock about that package or plugin. There is only some information in vendor folder.

I don't want to be asked anything about that plugin when I switch to a different branch and execute composer install. Composer.lock file exists in a project and not ignored.

I would appriciate for any explanation or help regarding it. 
Thank youi in advance.

**Comment by stof** [comment, marked as answer]
Plugins gets loaded based on the installed packages. And by the time you switch branch, this plugin is still installed in your vendor folder (until you run an install that would uninstall that package).

**Reply by s-shpachynskyi** [reply]
Thank you for explanation.
Isn't it logically just to uninstall plugin which doesn't exist anymore, I mean after I run composer install? Or maybe there's a way to skip such plugins, so I won't be asked during installation

**Reply by Seldaek** [reply]
Yes if you run composer install it will uninstall the plugin. But the plugin is installed so we initialize it when composer boots. I think I agree that if it is not required anywhere it maybe shouldn't be activated tho.

**Reply by naderman** [reply]
I agree with the above (should output a warning though), as uninstall is not very reliable anyway (e.g. vendor dir just gets deleted by user, or a composer command is run with --no-plugins in between other commands) the plugin should probably be resilient enough to work under these circumstances anyway.

**Reply by Seldaek** [reply]
Here is a PR fixing this https://github.com/composer/composer/pull/12000
